### PR TITLE
Filter solution information in quizzes after the first submission was done

### PR DESCRIPTION
**Pull request by TobiasPr**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) locally

### Motivation and Context

Student should not get information about the correct quiz before the exam is over

### Description

Filter informations from StudentExam before the exam is over - after the last student submitted -> the information is included again, so that it can be correctly displayed when the student sees the review

### Steps for Testing

1. Participate in an exam with MC, DnD and SA questions
2. After the first save of the questions, reload the exam
3. Check the server response for information about the solution -> you shouldn't find any information

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java [hunk 1]
```diff
@@ -174,12 +177,12 @@ public StudentExamResource(ExamAccessService examAccessService, StudentExamServi
      * GET /courses/{courseId}/exams/{examId}/studentExams/conduction : Find a student exam for the user.
      * This will be used for the actual conduction of the exam. The student exam will be returned with the exercises
      * and with the student participation and with the submissions.
-     *
+     * <p>
```
Comment (sascha11110) [inline src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java:180]: ?

File: src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java [hunk 2]
```diff
@@ -151,15 +154,15 @@ public StudentExamResource(ExamAccessService examAccessService, StudentExamServi
     /**
      * POST /courses/{courseId}/exams/{examId}/studentExams/submit : Submits the student exam
      * Updates all submissions and marks student exam as submitted according to given student exam
-     *
+     * <p>
```
Comment (sascha11110) [inline src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java:157]: ?

Comment (fde312) [inline src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java:157]: 🙉

File: src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java (hunk 1 above)
Comment (fde312) [inline src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java:180]: 🙉

File: src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java [hunk 3]
```diff
@@ -273,6 +281,22 @@ public void filterForExam(Exercise exercise, List<StudentParticipation> particip
                 if (result != null) {
                     participation.setResults(Set.of(result));
                 }
+                // for quiz exercises remove correct answer options / mappings from response
+                if (ZonedDateTime.now().isBefore(latestEndDate) && exercise instanceof QuizExercise) {
+                    QuizSubmission quizSubmission = (QuizSubmission) latestSubmission.get();
+                    quizSubmission.getSubmittedAnswers().forEach(submittedAnswer -> {
+                        QuizQuestion question = submittedAnswer.getQuizQuestion();
+                        if (question instanceof ShortAnswerQuestion) {
+                            ((ShortAnswerQuestion) question).setCorrectMappings(null);
```
Comment (fde312) [inline src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java:290]: We need to remove also the solutions from short answer questions:
image

Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java]: Not sure where exactly you saw this. I just tried it out and the solutions are not in the response any more

**Comment by krusche** [comment]
Good implementation 👍  Thank you! I only changed it by using the publish result date instead of the latestEndDate. This is immediately available and more secure, because the solution should not be "leaked" after the end of the exam

**Pull request merged**